configurator/src/main.rs

```diff
@@ -623,123 +656,120 @@ fn main() -> Result<(), anyhow::Error> {
+                                .lines()
+                                .filter(|l| !l.is_empty())
+                                .filter(|l| *l != "-----BEGIN CERTIFICATE-----")
+                                .filter(|l| *l != "-----END CERTIFICATE-----")
+                                .collect::<String>()
+                        )?,
+                        base64::Config::new(base64::CharacterSet::UrlSafe, false)
+                    ),
+                    macaroon = base64::encode_config(
+                        &macaroon_vec,
+                        base64::Config::new(base64::CharacterSet::UrlSafe, false)
+                    ),
+                ),
+                description: Some(
+                    "Use this for other applications that require a gRPC connection".to_owned(),
+                ),
+                copyable: true,
+                qr: true,
+                masked: true,
+            };
+            let lnd_connect_rest = Property {
+                value_type: "string".to_owned(),
+                value: format!(
+                    "lndconnect://{control_tor_address}:8080?cert={cert}&macaroon={macaroon}",
+                    control_tor_address = control_tor_address,
+                    cert = base64::encode_config(
+                        base64::decode(
+                            tls_cert
+                                .lines()
+                                .filter(|l| !l.is_empty())
+                                .filter(|l| *l != "-----BEGIN CERTIFICATE-----")
+                                .filter(|l| *l != "-----END CERTIFICATE-----")
+                                .collect::<String>()
+                        )?,
+                        base64::Config::new(base64::CharacterSet::UrlSafe, false)
+                    ),
+                    macaroon = base64::encode_config(
+                        &macaroon_vec,
+                        base64::Config::new(base64::CharacterSet::UrlSafe, false)
+                    ),
+                ),
+                description: Some(
+                    "Use this for other applications that require a REST connection".to_owned(),
+                ),
+                copyable: true,
+                qr: true,
+                masked: true,
+            };
+            let node_uri = Property {
+                value_type: "string".to_owned(),
+                value: format!(
+                    "{pubkey}@{peer_tor_address}:9735",
+                    pubkey = ni.identity_pubkey,
+                    peer_tor_address = peer_tor_address
+                ),
+                description: Some(
+                    "Give this to others to allow them to add your LND node as a peer".to_owned(),
+                ),
+                copyable: true,
+                qr: true,
+                masked: false,
+            };
+            let stats = Properties {
                 version: 2,
-                data: Data {
+                data: Data::LND{
```

Comment:
you want to just create a struct here and then render it out to properties as a separate step I think.

Reply:
Moved to P2 https://github.com/Start9Labs/lnd-wrapper/issues/41